### PR TITLE
feat: Email made translatable

### LINKED ISSUE #7405: Make emails translatable
EMails are defined in https://github.com/fossasia/open-event-server/blob/development/app/api/helpers/system_mails.py

They should be translatable and extracted in the POT files

### PR DESCRIPTION
**Pull request by codedsun**
Fixes #7405 

#### Changes proposed in this pull request:

-
-
-

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The unit tests pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

- [x] All the functions created/modified in this PR contain relevant docstrings.

### CODE REVIEW

File: app/api/auth.py [hunk 1]
```diff
@@ -342,19 +343,23 @@ def reset_password_post():
...
-                user,
-                PASSWORD_RESET_AND_VERIFY,
-                app_name=get_settings()['app_name'],
+            context = dict(
                 link=link,
             )
+            send_email(
+                to=user,
+                action=PASSWORD_RESET_AND_VERIFY,
+                html=render_template('email/password_reset_and_verify', **context),
```
Comment (iamareebjamal) [inline app/api/auth.py:352]: pass context as keyword arguments

File: app/api/auth.py [hunk 2]
```diff
@@ -342,19 +343,23 @@ def reset_password_post():
...
-                user,
-                PASSWORD_RESET,
-                app_name=get_settings()['app_name'],
-                link=link,
-                token=user.reset_password,
+            context = dict(link=link, settings=get_settings(), token=user.reset_password)
+            send_email(
+                to=user,
+                action=PASSWORD_RESET,
+                html=render_template('email/password_reset', **context),
```
Comment (iamareebjamal) [inline app/api/auth.py:361]: pass context as keyword arguments

File: app/api/users.py [hunk 3]
```diff
@@ -93,13 +94,17 @@ def after_create_object(self, user, data, view_kwargs):
...
             email=user.email,
             link=link,
-            frontend_url=settings['frontend_url'],
+            settings=get_settings(),
+        )
+        send_email(
+            to=user,
+            action=USER_REGISTER,
+            subject=MAILS[USER_REGISTER]['subject'].format(app_name=settings['app_name']),
+            html=render_template('email/user_register.html', **context),
```
Comment (iamareebjamal) [inline app/api/users.py:106]: pass context as keyword arguments

File: app/api/auth.py [hunk 4]
```diff
@@ -413,8 +418,14 @@ def change_password():
...
             save_to_db(user)
-            send_email_with_action(
-                user, PASSWORD_CHANGE, app_name=get_settings()['app_name']
+            send_email(
+                to=user,
+                action=PASSWORD_CHANGE,
+                subject=MAILS[PASSWORD_CHANGE]['subject'].format(
+                    app_name=get_settings()['app_name']
+                ),
+                html=render_template('email/password_change'),
```
Comment (iamareebjamal) [inline app/api/auth.py:427]: .html missing

File: app/api/auth.py (hunk 2 above)
Comment (iamareebjamal) [reply app/api/auth.py:361]: .html missing

File: app/api/auth.py (hunk 1 above)
Comment (iamareebjamal) [reply app/api/auth.py:352]: .html missing

File: app/templates/email/password_reset.html [hunk 5]
```diff
@@ -0,0 +1,3 @@
+Please use the following link to reset your password.
+<br/><a href='{link}' target='_blank'>{{ link }}</a>
```
Comment (iamareebjamal) [inline app/templates/email/password_reset.html:2]:
```suggestion
<br/>{{ link }}
```

File: app/templates/email/password_reset_and_verify.html [hunk 6]
```diff
@@ -0,0 +1,2 @@
+Please use the following link to reset your password and verify your account.
+<br> <a href='{link}' target='_blank'> {{ link }} </a>
```
Comment (iamareebjamal) [inline app/templates/email/password_reset_and_verify.html:2]:
```suggestion
<br>{{ link }}
```

**Comment by codedsun** [comment]
@iamareebjamal - mo file created.

File: app/templates/email/ticket_purchased.html [hunk 7]
```diff
@@ -1,19 +1,19 @@
-Hello,
-<br/><br/>Thank you for your ticket order for {{ order.event.name }}.
-<br/><br/>This is your order confirmation. You can download your tickets at: {{ order_view_url }}
-<br/><br/>Order Summary:
+{{ _('Hello') }},
+<br/><br/>{{ _('Thank you for your ticket order for') }} {{ order.event.name }}.
+<br/><br/>{{ _('This is your order confirmation. You can download your tickets at') }} : {{ order_view_url }}
+<br/><br/>{{ _('Order Summary') }}:
 <br/>Order: {{ order.invoice_number }}
 <br/>Name: {{ order.user.full_name }}
```
Comment (iamareebjamal) [inline app/templates/email/ticket_purchased.html:6]: _() missing here

File: app/templates/email/ticket_purchased_attendee.html [hunk 8]
```diff
@@ -1,17 +1,17 @@
-Hello,
-<br/><br/>This is a confirmation mail of your tickets for the event {{ order.event.name }}. You can download your tickets at: {{ order_view_url }}
-<br/><br/>Ticket Summary:
+{{ _('Hello') }},
+<br/><br/>{{ _('This is a confirmation mail of your tickets for the event') }} {{ order.event.name }}. {{ _('You can download your tickets at') }}: {{ order_view_url }}
+<br/><br/>{{ _('Ticket Summary') }}:
 <br/>Name: {{ attendees.0.user.full_name }}
```
Comment (iamareebjamal) [inline app/templates/email/ticket_purchased_attendee.html:4]: _() missing here

File: app/templates/email/user_register.html [hunk 9]
```diff
@@ -0,0 +1,7 @@
+{{ _('Hello') }},
+<br/><br/>{{ _('Your account has been created on') }} {{ settings.app_name }}. {{ _('Congratulations! ') }}
```
Comment (iamareebjamal) [inline app/templates/email/user_register.html:2]:
```suggestion
<br/><br/>{{ _('Your account has been created on') }} {{ settings.app_name }}. {{ _('Congratulations!') }}
```

File: app/templates/email/user_register.html [hunk 10]
```diff
@@ -0,0 +1,7 @@
...
+<br/><br/>{{ _('Your login is: {{ email }} ')}}
```
Comment (iamareebjamal) [inline app/templates/email/user_register.html:3]: email should be out of _()

File: app/templates/email/user_register.html [hunk 11]
```diff
@@ -0,0 +1,7 @@
...
+<br/><br/>{{ _('Please visit the following link to verify your email:') }} {{ link }}
+<br/><br/>{{ _('Thank You') }},
+<br/><br/>{{ settings.app_name }} Team
```
Comment (iamareebjamal) [inline app/templates/email/user_register.html:6]: _() missing here

**Comment by codedsun** [comment]
@iamareebjamal - While passing context as keyword argument in send_email function , the jinja html template doesn't recognize values and give error object not found.

**Comment by iamareebjamal** [comment]
You don't pass context as kwarg, you pass contents of context as kwargs

File: app/api/users.py [hunk 12]
```diff
@@ -93,13 +94,17 @@ def after_create_object(self, user, data, view_kwargs):
...
             link=link,
-            frontend_url=settings['frontend_url'],
+            settings=get_settings(),
+        )
+        send_email(
+            to=user,
+            action=USER_REGISTER,
+            subject=MAILS[USER_REGISTER]['subject'].format(app_name=settings['app_name']),
+            html=render_template('email/user_register.html', context=context),
+            attachments=None,
```
Comment (iamareebjamal) [inline app/api/users.py:107]: If attachments is None, then this is not needed

File: config.py [hunk 13]
```diff
@@ -49,6 +49,7 @@ class Config:
     FLASK_ADMIN_SWATCH = 'lumen'
 
     VERSION = VERSION_NAME
+    LANGUAGES = ['en', 'de', 'fr']
```
Comment (iamareebjamal) [inline config.py:52]: Add all languages present in the translations folder

File: babel.cfg [hunk 14]
```diff
@@ -0,0 +1,2 @@
+[jinja2: app/templates/**.html]
```
Comment (iamareebjamal) [inline babel.cfg:1]: Add python files here as well

File: messages.pot [hunk 15]
```diff
@@ -0,0 +1,125 @@
+# Translations template for PROJECT.
```
Comment (iamareebjamal) [inline messages.pot:1]: Delete this file

**Review by iamareebjamal — CHANGES_REQUESTED** [review]
Add required dependencies in requirements

**Comment by codedsun** [comment]
@iamareebjamal - Done the changes

File: requirements/common.txt [hunk 16]
```diff
@@ -62,6 +62,7 @@ pyyaml==5.3.1
 sendgrid==6.4.7
 marshmallow==2.15.2
 WeasyPrint==52.1
+Flask-Babel=2.0.0
```
Comment (iamareebjamal) [inline requirements/common.txt:65]: wrong syntax

**Comment by iamareebjamal** [comment]
Please add a screenshot of the mail you received and fix the build

File: requirements/common.txt (hunk 16 above)
Comment (iamareebjamal) [inline requirements/common.txt:65]:
```suggestion
Flask-Babel==2.0.0
```

**Comment by codedsun** [comment]
@iamareebjamal For every mail we save the text in the `mails` table. Can  i show you what was present there under column message.

```
Hello,
<br/><br/>Sunet Srivastava Event Till March.
<br/><br/>This is your order confirmation. You can download your tickets at : http://eventyay.com/orders/edd5f2a3-c44e-44fd-905b-4019ad975b4f/view
<br/><br/>Order Summary:
<br/>Order: O1604747108-22
<br/>Name: User User
<br/>

    <br/>'Ticket': Free Ticket
    <br/> 'Suneet': 1
    <br/>

<br/>About this event :
<br/>Datum: October 31, 2020 02:10 PM - March 01, 2021 05:00 PM
<br/>Online
<br/><br/>Best regards,
<br/>Open Event Team
<br/>--
<br/>Login to manage your orders at http://eventyay.com
```

**Comment by iamareebjamal** [comment]
That's in English

**Comment by iamareebjamal** [comment]
Build is still feeling. Seems like you didn't check the failing build I mentioned

**Comment by codedsun** [comment]
> Build is still feeling. Seems like you didn't check the failing build I mentioned

I checked, i thought it would be fixed but still failing. Let me fix it again

**Comment by codedsun** [comment]
@iamareebjamal pytype ci also needs to be fixed?

**Comment by iamareebjamal** [comment]
Yes

**Comment by codedsun** [comment]
@iamareebjamal - need your help for pytype issue, i am unable to resolve this. i am not getting what's the issue.

**Comment by iamareebjamal** [comment]
Disable https://google.github.io/pytype/errors.html#mro-error on `app/instance.py", line 243, in get_locale`

**Comment by iamareebjamal** [comment]
Please also write the command to extract to pot, merge to po, and generating mo files because I'll need that for testing

**Comment by codedsun** [comment]
Extracting to pot
`pybabel extract -F babel.cfg -k _l -o app/translations/messages.pot .`

Updating po
` pybabel update -i app/translations/messages.pot -d app/translations`

Generating mo
`pybabel compile -d app/translations`

**Comment by codedsun** [comment]
@iamareebjamal - Done!

**Comment by iamareebjamal** [comment]
How did you test the translation was happening?

**Comment by codedsun** [comment]
> How did you test the translation was happening?

As you sugggested, In the `de` folder I changed the msgStr of Ticket purchased string `Thank you for your ticket order for` to `Suneet Srivastava` 

i changed my browser display language in firefox to German (De), the website was translated to german now. 
I logged in and purchased a ticket. Now in the mails table I gave you the exact text of the message here. 
https://github.com/fossasia/open-event-server/pull/7416#issuecomment-723538229

**Comment by iamareebjamal** [comment]
OK. Thank you

File: app/templates/email/ticket_purchased_attendee.html [hunk 17]
```diff
@@ -1,17 +1,17 @@
...
+<br/><br/>{{ _('This is a confirmation mail of your tickets for the event') }} {{ order.event.name }}. {{ _('You can download your tickets at') }}: {{ order_view_url }}
+<br/><br/>{{ _('Ticket Summary') }}:
+<br/>{{ _('Name') }}: {{ attendees.0.user.full_name }}
 <br/>
-{% for _, attendee in attendees | groupby('ticket_id') %}
-    <br/>Ticket: {{ attendee.0.ticket.name }}
-    <br/>Quantity: {{ attendee | length }}
+{% for i, attendee in attendees | groupby('ticket_id') %}
+    <br/>{{ _('Ticket') }}: {{ attendee.0.ticket.name }}
+    <br/>{ _('Quantity') }}: {{ attendee | length }}
```
Comment (iamareebjamal) [inline app/templates/email/ticket_purchased_attendee.html:8]: { missing

**Review by iamareebjamal — APPROVED** [review]

**Pull request merged**